src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -265,72 +265,100 @@ public UnspentCoinModel[] ListUnspent(int minConfirmations = 1, int maxConfirmat
-                    {
-                        ScriptPubKey = BitcoinAddress.Create(address.Key, this.fullNode.Network).ScriptPubKey,
-                        Amount = Money.Coins(address.Value),
-                        SubtractFeeFromAmount = subtractFeeFromAddresses == null ? false : subtractFeeFromAddresses.Contains(BitcoinAddress.Create(address.Key, this.fullNode.Network))
-                    });
-                }
+                addresses = JsonConvert.DeserializeObject<Dictionary<string, decimal>>(addressesJson);
             }
             catch (JsonSerializationException ex)
             {
                 throw new RPCServerException(RPCErrorCode.RPC_PARSE_ERROR, ex.Message);
             }
 
-            if (recipients == null || !recipients.Any())
-            {
+            if (addresses.Count == 0)
                 throw new RPCServerException(RPCErrorCode.RPC_INVALID_PARAMETER, "No valid output addresses specified.");
+
+            // Optional list of addresses to subtract fees from.
+            IEnumerable<BitcoinAddress> subtractFeeFromAddresses = null;
+            if (!string.IsNullOrEmpty(subtractFeeFromJson))
+            {
+                try
+                {
+                    subtractFeeFromAddresses = JsonConvert.DeserializeObject<List<string>>(subtractFeeFromJson).Select(i => BitcoinAddress.Create(i, this.fullNode.Network));
+                }
+                catch (JsonSerializationException ex)
+                {
+                    throw new RPCServerException(RPCErrorCode.RPC_PARSE_ERROR, ex.Message);
+                }
+            }
+
+            var recipients = new List<Recipient>();
+            foreach (var address in addresses)
+            {
+                // Check for duplicate recipients
+                var recipientAddress = BitcoinAddress.Create(address.Key, this.fullNode.Network).ScriptPubKey;
+                if (recipients.Any(r => r.ScriptPubKey == recipientAddress))
+                    throw new RPCServerException(RPCErrorCode.RPC_INVALID_PARAMETER, string.Format("Invalid parameter, duplicated address: {0}.", recipientAddress));
+
+                var amount = Money.Coins(address.Value);
+                if (amount <= 0)
+                    throw new RPCServerException(RPCErrorCode.RPC_TYPE_ERROR, "Invalid amount to send.");
+
+                var recipient = new Recipient
+                {
+                    ScriptPubKey = recipientAddress,
+                    Amount = amount,
+                    SubtractFeeFromAmount = subtractFeeFromAddresses == null ? false : subtractFeeFromAddresses.Contains(BitcoinAddress.Create(address.Key, this.fullNode.Network))
+                };
+
+                recipients.Add(recipient);
             }
 
             var accountReference = this.GetAccount();
+            var spendable = this.walletManager.GetSpendableTransactionsInAccount(accountReference, minConf);
```

Comment:
Is this check not made inside the create trx?

Reply:
Double checked, negative amount will throw an ArgumentOutOfRange exception in wallet. Duplicate recipient is an interesting one, I don't think the wallet cares about that as it's valid to send to the same address more than once (a bit dumb but valid). So I guess we can do this check in the RPC if core does.

Reply:
> Double checked, negative amount will throw an ArgumentOutOfRange exception in wallet. 

OK sweet, I will remove this check then 👍